### ISSUE TITLE
Setting a featured image and clicking Preview does not show the featured image

### ISSUE THREAD

**Issue by noisysocks** [WordPress/gutenberg#9151]
**Describe the bug**
After you set a feature image, clicking _Preview_ does not show the feature image on the frontend. This is a very confusing experience.

**To Reproduce**
1. Create a new post
2. Insert a title or add some text
3. Using the Document sidebar, set a _Featured Image_
4. Click _Preview_ in the top right

**Expected behavior**
The selected featured image should be visible at the top of the post, but it is not.

**Screenshots**
![featured-image](https://user-images.githubusercontent.com/612155/44322905-f3f68500-a492-11e8-9d64-9e70614231a5.gif)

**Additional context**
Encountered this during user testing. Clicking _Save Draft_ before _Preview_ works as expected.

**Comment by noisysocks** [comment]
Is this because the autosave endpoint does not accept the `featured_media` field? cc. @azaozz @adamsilverstein @danielbachhuber

**Comment by danielbachhuber** [comment]
The underlying issue is that the Classic Editor saves _all_ post meta during a Preview request. However, Gutenberg only saves a few specific fields.

I think https://core.trac.wordpress.org/ticket/20299 is defacto Trac ticket on the subject.

I'm not sure there's an "easy fix" for this one. We have some liberty on how we can address the issue though. I don't have strong opinions on the preferred implementation at this point.

**Comment by youknowriad** [comment]
Also like suggested here https://github.com/WordPress/gutenberg/issues/7767#issuecomment-406397225 The fact that the classic edditor savese those metas is considered a bug, because they are not saved as a "revision" but they really update the post object while the intent is to only preview and not save the changes. 

I closed the other issue as Invalid, I think we should dod the same.

**Comment by designsimply** [comment]
Are you suggesting to close the issue as invalid because you don't think the ability to preview changes to featured images should be doable or because the problem is technically happening due to an upstream bug in the way that autosaves work?

**Comment by youknowriad** [comment]
Because I don't think it's doable without a very big refactoring to how revisions work in WordPress and the fact that it works in the classic editor is just a consequence of a bug in the classic editor.

**Comment by danielbachhuber** [comment]
We could potentially save all post meta _only_ to the autosave object, which would mitigate @azaozz's original concerns about bloating revisions with post meta (and why revisions meta, to my knowledge, wasn't ever implemented).

**Comment by danielbachhuber** [comment]
> We could potentially save all post meta _only_ to the autosave object

Notably, if we do this, we'd need to make sure the UX around restoring from an autosave is updated accordingly. Either the entire autosaved object, including meta, is restored, or only the title, content, and excerpt are restored from the autosave object, even though meta is present too.

**Comment by noisysocks** [comment]
I don't think we should close the issue—it came up during user testing and was a considerable roadblock for them using Gutenberg. 

In the Classic Editor if I set a featured image and then click _Preview Changes_ it correctly previews the featured image and does **not** update the published post.

**Comment by youknowriad** [comment]
> The tl;dr: for draft posts we would need to do "save draft" before showing the preview, i.e. save all post data including from "metaboxes".

I know this was the case at some point, so I'm wondering if there's a valid reason for its change to "auto-save". cc @aduth

**Comment by aduth** [comment]
Previously: #7130

The client wants to create an autosave, it shouldn't care who's authored the post, whether it's a draft. This should be on the server to handle. Otherwise what purpose is the endpoint serving? We can send the entire payload, but if we're saying the prerequisite for a preview is an autosave, the incidentals should be contained within the handling of the autosave task on the server.

**Comment by danielbachhuber** [comment]
Given the [timeline for shipping 5.0](https://make.wordpress.org/core/2018/10/31/wordpress-5-0-schedule-updates/), it's unlikely this particular bug will be fixed in this release. Reassigning to `WP 5.1 Onwards` to pick up in the future.

**Comment by noisysocks** [comment]
Thoughts on borrowing what the Classic Editor does as an interim solution for 5.0?

> In the Classic Editor if I set a featured image and then click _Preview Changes_ it correctly shows a preview of the post with a featured image and does **not** update the published post. It looks like it does this by [sending `_thumbnail_id` as a query variable to the preview screen](https://core.trac.wordpress.org/changeset/38118), which is maybe an interim solution that we should borrow for Gutenberg.

**Comment by noisysocks** [comment]
Hm, damn. [I tried it](https://github.com/WordPress/gutenberg/compare/try/feature-image-preview) but it only seems to fix the case where you're previewing changes made to a published post.

**Comment by danielbachhuber** [comment]
Yeah, it looks like this is more severe (#11280) and will need to be addressed in some form.

**Comment by adamsilverstein** [comment]
Noting that in my testing, this is **not** fixed by https://github.com/WordPress/gutenberg/pull/11409 which saves metabox meta before previewing. I think we can make passing the thumbnail id work, this looks like a good start: https://github.com/WordPress/gutenberg/compare/try/feature-image-preview

**Comment by adamsilverstein** [comment]
@noisysocks your try branch looks good and solves the issue for published posts - to me this seems worth merging as is. We can resolve the issue with draft posts separately.
